.github/workflows/publish.yml

```diff
@@ -40,11 +36,37 @@ jobs:
       - name: Build website
         run: make clean && make publish
 
-      - name: Commit output to gh-pages branch
+      - name: Fix permissions if needed
         run: |
-            git config user.name "Automated"
-            git config user.email "actions@users.noreply.github.com"
-            timestamp=$(date -u +%FT%T%z)
-            ghp-import -m "Generate Pelican site ${timestamp}" --no-history --cname 'conference.scipy.org' --branch gh-pages  output
-            git push --force origin gh-pages
-        
+          chmod -c -R +rX "output/" | while read line; do
+            echo "::warning title=Invalid file permissions automatically fixed::$line"
+          done
+
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: 'output/'
+
+  deploy:
+    name: Deploy to GitHub Pages
+    # Deploy on push event to main or with any branch and workflow dispatch
+    if: (github.event_name == 'push' && github.ref == 'refs/heads/main') || github.event_name == 'workflow_dispatch'
+    needs: build
+    # Set permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+    permissions:
+      pages: write      # to deploy to Pages
+      id-token: write   # to verify the deployment originates from an appropriate source
+
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
```

Comment:
c.f. https://github.com/actions/deploy-pages?tab=readme-ov-file#usage again for why these choices but make sure to switch to using GitHub Actions under https://github.com/scipy-conference/conference.scipy.org/settings/pages

![image](https://github.com/scipy-conference/conference.scipy.org/assets/5142394/7c9dd66a-e3ac-4557-8e48-4438c908dabe)

and then set whatever branch protection rules you'd like for the `github-pages` environment under https://github.com/scipy-conference/conference.scipy.org/settings/environments